modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/AddRapidSpecies.pm

```diff
@@ -0,0 +1,120 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::AddRapidSpecies
+
+=head1 DESCRIPTION
+
+Runnable to add species as a new genome_db directly into pipeline database from core.
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::AddRapidSpecies;
+
+use warnings;
+use strict;
+use List::Util qw(any);
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Utils::Argument qw(rearrange);
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::Compara::Utils::MasterDatabase;
+use Data::Dumper;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+sub param_defaults {
+    return {
+        'force'      => 1,
+        'do_not_add' => undef,
+        'division'   => 'all',
+    };
+}
+
+sub fetch_input {
+    my ($self) = @_;
+
+    my $species_list_file = $self->param_required('species_list_file');
+    my $do_not_add        = $self->param('do_not_add');
```

Comment:
Yes, this is here incase what comes in through the `species_list_file` also includes the reference genomes. Prior to meeting on Thursday, it was unknown how the species would be passed through. I have not had the chance to properly investigate the runnable `SpeciesFactory` from production yet, but do we know that this will not pass through any of the reference genomes? The alternative (for next PR if there is the possibility that references could be picked up) would be to check back to the reference master database for the species to exclude?